controller/setting_controller.go

```diff
@@ -914,6 +922,38 @@ func (sc *SettingController) updateKubernetesClusterAutoscalerEnabled() error {
 	return nil
 }
 
+func (sc *SettingController) cleanupShareManagerServiceAndEndpoints() error {
+	var err error
+	defer func() {
+		if err != nil {
+			err = errors.Wrapf(err, "failed to cleanup share manager service and endpoints for %s setting update", types.SettingNameStorageNetworkForRWXVolumeEnabled)
+		}
+	}()
+
+	shareManagers, err := sc.ds.ListShareManagers()
+	if err != nil {
+		return err
+	}
+
+	for _, shareManager := range shareManagers {
+		_log := sc.logger.WithField("shareManager", shareManager.Name)
+
+		_log.WithField("service", shareManager.Name).Infof("Deleting Service for %v setting update", types.SettingNameStorageNetworkForRWXVolumeEnabled)
+		err := sc.ds.DeleteService(shareManager.Namespace, shareManager.Name)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		_log.WithField("endpoint", shareManager.Name).Infof("Deleting Endpoint for %v setting update", types.SettingNameStorageNetworkForRWXVolumeEnabled)
+		err = sc.ds.DeleteKubernetesEndpoint(shareManager.Namespace, shareManager.Name)
```

Comment:
Was it an oversight from before that we did not delete the Endpoints? Or does this PR somehow require us to do so now when the last one did not?

Reply:
The Endpoint is owned by the ShareManager, so it gets cleaned up when the ShareManager is removed. Because it gets cleaned up during my test, so I overlooked that we should also clean up the Endpoint when the Service is gone. It's a minor, non-blocking bug.